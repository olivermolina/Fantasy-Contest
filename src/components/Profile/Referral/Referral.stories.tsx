import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import ReferralMenu from '.';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Lockspread/Profile',
  component: ReferralMenu,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as ComponentMeta<typeof ReferralMenu>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof ReferralMenu> = (args) => (
  <ReferralMenu {...args} />
);

export const Referral = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Referral.args = {
  referralCode: 'jsmith321',
  referralCustomText:
    'Refer a friend from the link or code in your profile and get $25 in bonus credit!',
};
