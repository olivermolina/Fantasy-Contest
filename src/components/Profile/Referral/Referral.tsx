import React, { useState } from 'react';
import Icons from '~/components/Icons/Icons';
import { Tooltip } from '@mui/material';

interface CopyButtonProps {
  handleCopy: () => void;
}

const CopyButton = (props: CopyButtonProps) => {
  const [open, setOpen] = useState(false);
  const handleClick = () => {
    props.handleCopy?.();
    setOpen(true);
  };
  const handleClose = () => {
    setOpen(false);
  };
  return (
    <Tooltip title="Copied to clipboard!" open={open} onClose={handleClose}>
      <button onClick={handleClick}>
        <Icons.ClipboardDocument className={'h-6 w-6'} />
      </button>
    </Tooltip>
  );
};

export interface ReferralProps {
  /**
   * Referral Code
   */
  referralCode: string;
  /**
   * Referral Description
   */
  referralCustomText: string;
}

const styles = {
  item: 'flex flex-col rounded-lg gap-2',
  field:
    'flex justify-start items-center bg-[#2F5F98] rounded-lg divide-x divide-slate-500 border-slate-500 border',
  text: 'p-2 bg-[#1A487F] w-full rounded-r-lg',
  copyButton: 'bg-[#2F5F98] rounded-l-lg px-1',
};

const Referral = ({ referralCode, referralCustomText }: ReferralProps) => {
  const referralLink = `${window.location.origin}/auth/sign-up?referral=${referralCode}`;
  return (
    <div className={`w-full lg:p-4`}>
      <div className="flex flex-col p-2 lg:p-6 gap-2 rounded-lg shadow-md gap-4">
        <p className="font-bold text-xl">Referral</p>
        <p className="font-semibold text-md">{referralCustomText}</p>
        <div
          className={`flex flex-col lg:flex-row justify-start gap-4 lg:gap-10`}
        >
          <div className={styles.item}>
            <p>Referral Code</p>
            <div className={styles.field}>
              <div className={styles.copyButton}>
                <CopyButton
                  handleCopy={() => navigator.clipboard.writeText(referralCode)}
                />
              </div>
              <p className={styles.text}>{referralCode}</p>
            </div>
          </div>
          <div className={styles.item}>
            <p>Referral Link</p>
            <div className={styles.field}>
              <div className={styles.copyButton}>
                <CopyButton
                  handleCopy={() => navigator.clipboard.writeText(referralLink)}
                />
              </div>
              <p className={styles.text}>{referralLink}</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Referral;
