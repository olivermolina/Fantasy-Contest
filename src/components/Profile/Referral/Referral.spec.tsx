import React from 'react';
import { render } from '@testing-library/react';
import Referral from './Referral';

describe('Referral', () => {
  it('should render correctly', () => {
    const props = {
      referralCode: 'jsmith321',
      referralCustomText:
        'Refer a friend from the link or code in your profile and get $25 in bonus credit!',
    };
    const { container } = render(<Referral {...props} />);
    expect(container).toMatchSnapshot();
  });
});
