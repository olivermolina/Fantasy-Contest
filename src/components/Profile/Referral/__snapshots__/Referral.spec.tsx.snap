// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Referral should render correctly 1`] = `
<div>
  <div
    class="w-full lg:p-4"
  >
    <div
      class="flex flex-col p-2 lg:p-6 gap-2 rounded-lg shadow-md gap-4"
    >
      <p
        class="font-bold text-xl"
      >
        Referral
      </p>
      <p
        class="font-semibold text-md"
      >
        Refer a friend from the link or code in your profile and get $25 in bonus credit!
      </p>
      <div
        class="flex flex-col lg:flex-row justify-start gap-4 lg:gap-10"
      >
        <div
          class="flex flex-col rounded-lg gap-2"
        >
          <p>
            Referral Code
          </p>
          <div
            class="flex justify-start items-center bg-[#2F5F98] rounded-lg divide-x divide-slate-500 border-slate-500 border"
          >
            <div
              class="bg-[#2F5F98] rounded-l-lg px-1"
            >
              <button
                aria-label="Copied to clipboard!"
                class=""
                data-mui-internal-clone-element="true"
              >
                <svg
                  class="h-6 w-6"
                  data-file-name="SvgClipboardDocument"
                />
              </button>
            </div>
            <p
              class="p-2 bg-[#1A487F] w-full rounded-r-lg"
            >
              jsmith321
            </p>
          </div>
        </div>
        <div
          class="flex flex-col rounded-lg gap-2"
        >
          <p>
            Referral Link
          </p>
          <div
            class="flex justify-start items-center bg-[#2F5F98] rounded-lg divide-x divide-slate-500 border-slate-500 border"
          >
            <div
              class="bg-[#2F5F98] rounded-l-lg px-1"
            >
              <button
                aria-label="Copied to clipboard!"
                class=""
                data-mui-internal-clone-element="true"
              >
                <svg
                  class="h-6 w-6"
                  data-file-name="SvgClipboardDocument"
                />
              </button>
            </div>
            <p
              class="p-2 bg-[#1A487F] w-full rounded-r-lg"
            >
              http://localhost/auth/sign-up?referral=jsmith321
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
